src/ts/components/DashSocketIO.tsx

```diff
@@ -1,64 +1,95 @@
 import React from 'react';
 import { io } from 'socket.io-client';
-import {DashComponentProps} from '../props';
+import { DashComponentProps } from '../props';
+
+type SocketOptions = {
+  /** The socket.io endpoint path, defaults to '/socket.io' */
+  path?: string;
+  /** Whether to force a new connection */
+  forceNew?: boolean;
+  /** Timeout in milliseconds */
+  timeout?: number;
+  /** Whether to automatically connect on creation */
+  autoConnect?: boolean;
+  /** Additional query parameters */
+  query?: Record<string, any>;
+  /** Authentication data */
+  auth?: Record<string, any>;
+  [key: string]: any;
+};
 
 type Props = {
   /** The socket.io namespace url, defaults to window.location.origin */
-  url?: string
+  url?: string;
+  /** Socket.IO connection options */
+  options?: Partial<SocketOptions>;
   /** Name of Socket.IO events to listen to */
-  eventNames?: string[]
+  eventNames?: string[];
   /** Whether the client is connected to the websocket - READONLY */
-  connected?: boolean
+  connected?: boolean;
   /** The socket ID */
-  socketId?: string
+  socketId?: string;
 } & DashComponentProps;
+
+const defaultOptions: SocketOptions = {
+  path: '/socket.io',
+  forceNew: false,
+  autoConnect: true
+};
+
 /**
  * Socket.IO Dash component.
  */
 const DashSocketIO = (props: Props) => {
-    const { url, setProps, eventNames } = props;
-
-    const socket = React.useMemo(() => io(
-      url || undefined, {auth: {pathname: window.location.pathname}}
-    ), [url])
+  const { url, options = {}, setProps, eventNames } = props;
```

Comment:
I'm not familiar with this syntax, will this make it an empty object only if it is null?

Reply:
It will destructure the options object. The default value is {}. If we have options {'timeout': 300} then we get that. If options: null, we would get null.